arkouda/array_view.py

```diff
@@ -0,0 +1,224 @@
+from arkouda.pdarrayclass import pdarray, parse_single_value, create_pdarray
+from arkouda.pdarraycreation import array, arange
+from arkouda.numeric import cumprod
+import numpy as np  # type: ignore
+from arkouda.client import generic_msg
+from arkouda.dtypes import translate_np_dtype, resolve_scalar_dtype
+from enum import Enum
+import json
+
+OrderType = Enum('OrderType', ['ROW_MAJOR', 'COLUMN_MAJOR'])
+
+
+class ArrayView:
+    def __init__(self, base: pdarray, shape, order='row_major'):
+        self.objtype = type(self).__name__
+        self.shape = array(shape)
+        if not isinstance(self.shape, pdarray):
+            raise TypeError(f"Shape cannot be type {type(self.shape)}")
+        if base.size != self.shape.prod():
+            raise ValueError(f"cannot reshape array of size {base.size} into shape {self.shape}")
+        self.base = base
+        self.size = base.size
+        self.ndim = self.shape.size
+        if order in {'C', 'row_major', 'ROW_MAJOR'}:
+            self.order = OrderType.ROW_MAJOR
+        elif order in {'F', 'column_major', 'COLUMN_MAJOR'}:
```

Comment:
May be a good idea to use `order.lower()` or `order.upper()` here. That way we cover things like Row_Major as well. Not entirely necessary though.

Reply:
good idea!